roles/common/tasks/iptables.yml

```diff
@@ -0,0 +1,76 @@
+---
+- name: Disable firewalld
+  service:
+    name: firewalld
+    state: stopped
+    enabled: false
+  failed_when: false
+
+- name: Install iptables
+  package: name=iptables
+  register: result
+  until: result|succeeded
+  retries: 5
+
+- name: Install iptables-services
+  package: name=iptables-services
+  register: result
+  until: result|succeeded
+  retries: 5
+
+- name: Enable iptables
+  service:
+    name: iptables
+    state: started
+    enabled: true
+
+- name: ensure iptables allows established and related traffic
+  lineinfile:
+    dest: /etc/sysconfig/iptables
+    state: present
+    regexp: "^.*INPUT.*ESTABLISHED,RELATED.*ACCEPT"
+    insertafter: "^:OUTPUT "
+    line: "-A INPUT -m state --state ESTABLISHED,RELATED -j ACCEPT"
+    backup: yes
+  notify: restart iptables
+
+- name: Permit SSH
+  lineinfile:
+    dest: /etc/sysconfig/iptables
+    state: present
+    regexp: "^.*INPUT.*{{ item.protocol }}.*{{ item.port }}.*ACCEPT"
+    insertbefore: "^-A INPUT -j REJECT --reject-with icmp-host-prohibited"
+    line: "-A INPUT -m state --state NEW -m {{ item.protocol }} -p {{ item.protocol }} --dport {{ item.port }} -j ACCEPT"
+    backup: yes
+  with_items:
+    - protocol: tcp
+      port: 22
+  notify: restart iptables
+
+- name: Permit SSH For Sources
+  lineinfile:
+    dest: /etc/sysconfig/iptables
+    state: present
+    regexp: "^.*INPUT.*{{ item.protocol }}.*-s {{ item.source }}.*{{ item.port }}.*ACCEPT"
+    insertbefore: "^-A INPUT -j REJECT --reject-with icmp-host-prohibited"
+    line: "-A INPUT -m state --state NEW -m {{ item.protocol }} -p {{ item.protocol }} -s {{ item.source }} --dport {{ item.port }} -j ACCEPT"
+    backup: yes
+  with_items:
+    - source: "{{ common.ssh.allow_from }}"
+      protocol: tcp
+      port: 22
+  notify: restart iptables
+
+- name: Permit unrestricted access from remainder of cluster
+  lineinfile:
+    dest: /etc/sysconfig/iptables
+    state: present
+    regexp: "^.*INPUT.*{{ item.protocol }}.*-s {{ item.source }}.*ACCEPT"
+    insertbefore: "^-A INPUT -j REJECT --reject-with icmp-host-prohibited"
+    line: "-A INPUT -m state --state NEW -m {{ item.protocol }} -p {{ item.protocol }} -s {{ item.source.cidr }} -j ACCEPT"
+    backup: yes
+  with_items:
+    - source: "{{ undercloud_cidr }}"
```

Comment:
It failed as well. My solution:

~~~
- name: Permit unrestricted access from remainder of cluster
  lineinfile:
    dest: /etc/sysconfig/iptables
    state: present
    regexp: "^.*INPUT.*{{ item[1] }}.*-s {{ item[0].cidr }}.*ACCEPT"
    insertbefore: "^-A INPUT -j REJECT --reject-with icmp-host-prohibited"
    line: "-A INPUT -m state --state NEW -p {{ item[1] }} -s {{ item[0].cidr }} -j ACCEPT"
    backup: yes
  with_nested:
    - "{{ undercloud_cidr }}"
    - all
  notify: restart iptables
  when: undercloud_cidr is defined
~~~